### PR TITLE
Made the batch normalization layer backend agnostic

### PR DESCRIPTION
**Pull request by asmith26**
Following the 'default' method within the [image preprocessing layer](https://github.com/fchollet/keras/blob/master/keras/preprocessing/image.py#L259-L261), I've tweaked the batch norm layer code to enable the 'axis' attribute to be inferred from `K.image_dim_ordering()`

### CODE REVIEW

**Comment by fchollet** [comment]
This would be incorrect in many cases (e.g. `dim_ordering` is irrelevant whenever the input being processed isn't a batch of images). I'd rather have the user explicitly set this argument. -1 is a reasonable default.
